File: builder/php/Lesson.php
<?php

namespace Builder;

// UML diyagramındaki Product sınıfına denk gelmektedir.
// Farklı sunumları olan sınıfımız budur.
class Lesson
{
    public int $id;
    public string $name;
    public float $price;
    public float $discountedPrice;
    public bool $discountedApplied;
    public string $lessonNote;
}

File: builder/go/lesson.go
package main

// UML diyagramındaki Product sınıfına denk gelmektedir.
// Farklı sunumları olan sınıfımız budur.
type lesson struct {
	id              int
	name            string
	price           float32
	discountedPrice float32
	discountApplied bool
	lessonNote      string
}

func newLesson() *lesson {
	return &lesson{}
}
